pandas/core/reshape.py

```diff
@@ -1213,7 +1216,7 @@ def make_axis_dummies(frame, axis='minor', transform=None):
         labels = cat.codes
         items = cat.categories
 
-    values = np.eye(len(items), dtype=float)
+    values = np.eye(len(items), dtype=np.uint8)
```

Comment:
technically you need something like _coerce_indexer_dtype from https://github.com/pydata/pandas/blob/master/pandas/types/cast.py

as u don't know if it will fit in a uint8 (u might need. 16 or 32)